base/special/exp.jl

```diff
@@ -68,71 +187,4 @@ Compute the natural base exponential of `x`, in other words ``e^x``.
 ```jldoctest
 julia> exp(1.0)
 2.718281828459045
-```
-"""
-exp(x::Real) = exp(float(x))
-function exp(x::T) where T<:Union{Float32,Float64}
-    xa = reinterpret(Unsigned, x) & ~sign_mask(T)
-    xsb = signbit(x)
-
-    # filter out non-finite arguments
-    if xa > reinterpret(Unsigned, MAX_EXP(T))
-        if xa >= exponent_mask(T)
-            xa & significand_mask(T) != 0 && return T(NaN)
-            return xsb ? T(0.0) : T(Inf) # exp(+-Inf)
-        end
-        x > MAX_EXP(T) && return T(Inf)
-        x < MIN_EXP(T) && return T(0.0)
-    end
-    # This implementation gives 2.7182818284590455 for exp(1.0) when T ==
-    # Float64, which is well within the allowable error; however,
-    # 2.718281828459045 is closer to the true value so we prefer that answer,
-    # given that 1.0 is such an important argument value.
-    if x == T(1.0) && T == Float64
-        return 2.718281828459045235360
-    end
-    # compute approximation
-    if xa > reinterpret(Unsigned, T(0.5)*T(LN2)) # |x| > 0.5 log(2)
-        # argument reduction
-        if xa < reinterpret(Unsigned, T(1.5)*T(LN2)) # |x| < 1.5 log(2)
-            if xsb
-                k = -1
-                hi = x + LN2U(T)
-                lo = -LN2L(T)
-            else
-                k = 1
-                hi = x - LN2U(T)
-                lo = LN2L(T)
-            end
-        else
-            n = round(T(LOG2_E)*x)
-            k = unsafe_trunc(Int,n)
-            hi = muladd(n, -LN2U(T), x)
-            lo = n*LN2L(T)
-        end
-        # compute approximation on reduced argument
-        r = hi - lo
-        z = r*r
-        p = r - z*exp_kernel(z)
-        y = T(1.0) - ((lo - (r*p)/(T(2.0) - p)) - hi)
-        # scale back
-        if k > -significand_bits(T)
-            # multiply by 2.0 first to prevent overflow, which helps extends the range
-            k == exponent_max(T) && return y * T(2.0) * T(2.0)^(exponent_max(T) - 1)
-            twopk = reinterpret(T, rem(exponent_bias(T) + k, uinttype(T)) << significand_bits(T))
-            return y*twopk
-        else
-            # add significand_bits(T) + 1 to lift the range outside the subnormals
-            twopk = reinterpret(T, rem(exponent_bias(T) + significand_bits(T) + 1 + k, uinttype(T)) << significand_bits(T))
-            return y * twopk * T(2.0)^(-significand_bits(T) - 1)
-        end
-    elseif xa < reinterpret(Unsigned, exp_small_thres(T)) # |x| < exp_small_thres
-        # Taylor approximation for small values: exp(x) ≈ 1.0 + x
-        return T(1.0) + x
-    else
-        # primary range with k = 0, so compute approximation directly
-        z = x*x
-        p = x - z*exp_kernel(z)
-        return T(1.0) - ((x*p)/(p - T(2.0)) - x)
-    end
-end
+""" exp(x::Real)
```

Comment:
Why delete `exp(x::Real) = exp(float(x))` ?

Reply:
It's defined within the loop above. (Line 138)